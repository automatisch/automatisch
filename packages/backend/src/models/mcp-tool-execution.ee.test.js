import { describe, it, expect } from 'vitest';
import McpTool from '@/models/mcp-tool.ee.js';
import McpToolExecution from '@/models/mcp-tool-execution.ee.js';
import Base from '@/models/base.js';

describe('McpToolExecution model', () => {
  it('tableName should return correct name', () => {
    expect(McpToolExecution.tableName).toBe('mcp_tool_executions');
  });

  it('jsonSchema should have correct validations', () => {
    expect(McpToolExecution.jsonSchema).toMatchSnapshot();
  });

  it('relationMappings should return correct associations', () => {
    const relationMappings = McpToolExecution.relationMappings();

    const expectedRelations = {
      mcpTool: {
        relation: Base.BelongsToOneRelation,
        modelClass: McpTool,
        join: {
          from: 'mcp_tool_executions.mcp_tool_id',
          to: 'mcp_tools.id',
        },
      },
    };

    expect(relationMappings).toStrictEqual(expectedRelations);
  });
});
