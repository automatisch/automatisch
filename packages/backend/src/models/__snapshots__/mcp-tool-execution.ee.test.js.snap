// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`McpToolExecution model > jsonSchema should have correct validations 1`] = `
{
  "properties": {
    "createdAt": {
      "type": "string",
    },
    "dataIn": {
      "type": [
        "object",
        "null",
      ],
    },
    "dataOut": {
      "type": [
        "object",
        "null",
      ],
    },
    "errorDetails": {
      "type": [
        "object",
        "null",
      ],
    },
    "id": {
      "format": "uuid",
      "type": "string",
    },
    "mcpToolId": {
      "format": "uuid",
      "type": "string",
    },
    "status": {
      "enum": [
        "success",
        "failure",
      ],
      "type": "string",
    },
    "updatedAt": {
      "type": "string",
    },
  },
  "type": "object",
}
`;
